chef/cookbooks/provisioner/recipes/bootdisk.rb

```diff
@@ -78,17 +79,42 @@
         node[:crowbar_wall][:boot_device] = "disk/by-id/#{bootdisk}"
       end
     end
+
     # If we still have a /dev/disk/by-path name in boot_device here and the
     # devices looks like a virtio device it is safer to use the base device
     # name for bootdisk instead of the by-path name. According to:
     #  https://www.mail-archive.com/systemd-devel@lists.freedesktop.org/msg20607.html
     # the by-path names for virtio device are not stable and don't even exist
-    # anymore with newer udev releases, e.g. on SLES12
+    # anymore with newer udev releases
+    # This is only needed for SLE_11_SP3 sleshammer. On SLE_12 based image 
+    # directory "disk/by-path" with virtio disks driver doesn't exist 
     if File.dirname(node[:crowbar_wall][:boot_device]) == "disk/by-path" and
         File.basename(node[:crowbar_wall][:boot_device]) =~ /-virtio-/
       Chef::Log.debug("#{node[:crowbar_wall][:boot_device]} appears to be a virtio device, falling back to simple device name #{dev}")
-      node[:crowbar_wall][:boot_device] = dev
+       node[:crowbar_wall][:boot_device] = dev
     end
+    
+    # Searching boot disk with virtio driver if no other disks were found
+    # on SLES_12 sleshammer image
+    basedir = "/sys/block"
+    device_driver = nil
+    if dev.nil?
+      ::Dir.entries(basedir).sort.each do |path|
+        next unless File.symlink?(File.join(basedir, path))
+        dev = File.readlink("#{basedir}/#{path}").split('/')[-1]
+        if dev =~ /^vd[a-z]+$/ && File.exists?("/sys/block/#{dev}/device/uevent")
+          File.open("/sys/block/#{dev}/device/uevent").each_line {
+            |line| device_driver = line.strip.split('=')[1] if line =~ /DRIVER/
```

Comment:
Style/MultilineBlockLayout: Block argument expression is not on the same line as the block start.<br>Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.